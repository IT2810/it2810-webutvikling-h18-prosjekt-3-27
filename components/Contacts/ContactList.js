import React, {Component} from "react";
import {FlatList, TouchableOpacity,} from "react-native";
import Contact from "./Contact";

export default class ContactList extends Component {
  constructor(props) {
    super(props);
    this.handlePress = this.handlePress.bind(this);
  }


  handlePress = (item) => {
  console.log(item);
  this.props.onContactPress(item);
};

  render () {
    return(
      <FlatList
        data={this.props.contacts}
        keyExtractor={contact => "" + contact.key}
        renderItem={({item}) =>
          <TouchableOpacity onPress={() => this.handlePress(item)}>
            <Contact
              contact={item}
            />
          </TouchableOpacity>}
      />
    )
  }
}

