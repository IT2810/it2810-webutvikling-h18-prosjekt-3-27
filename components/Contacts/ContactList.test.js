import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ContactList from "./ContactList";


test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<ContactList contacts={[]} selected={null}/>);
  expect(tree).toMatchSnapshot();
});