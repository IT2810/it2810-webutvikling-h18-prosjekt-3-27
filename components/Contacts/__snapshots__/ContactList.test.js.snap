// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<FlatList
  data={Array []}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
